variation-commons-core/src/main/java/uk/ac/ebi/eva/commons/core/models/ws/VariantWithSamplesAndAnnotation.java

```diff
@@ -51,6 +51,16 @@ public VariantWithSamplesAndAnnotation(String chromosome, long start, long end,
         sourceEntries = new HashMap<>();
     }
 
+    public VariantWithSamplesAndAnnotation(IVariant variant) {
+        super(variant.getChromosome(), variant.getStart(), variant.getEnd(), variant.getReference(), variant.getAlternate());
+        this.sourceEntries = new HashMap<>();
+        for (IVariantSourceEntry entry : variant.getSourceEntries()) {
+            VariantSourceEntryWithSampleNames entrySN = new VariantSourceEntryWithSampleNames(entry);
```

Comment:
Looking at the previous comment, we can see we need the samples name list for building a VariantSourceEntryWithSampleNames list (hence the class name). A 'List<String> sampleNames' parameter should be added to this constructor then. If we don't have the sample names to pass them to this constructor, we cannot use the VariantSourceEntryWithSampleNames class and we have to look for another alternative.

Reply:
Ok, I have checked, and this constructor is just used in the tests of the 'Spring Data Migration' eva-tools PR. As commented, this constructor is incomplete, we need the List of sample names for building VariantWithSamplesAndAnnotation objects. Add to this constructor this second parameter, and you can call then to     public VariantSourceEntryWithSampleNames(IVariantSourceEntry variantSourceEntry, List<String> sampleNames)  to build the VariantSource objects (so no changes in VariantSourceEntryWithSampleNames will be needed. In the tests, add List<String> sampleNames example objects, with invented sample names. Just be careful to add a sample Name for each sample info column